pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -133,6 +135,17 @@ func (s *Server) initDNSCerts() error {
 				}()
 				return nil
 			})
+		} else if _, err := os.Stat(path.Join(LocalCertDir.Get(), ca.IstioGenerated)); err == nil {
+			// TODO(jaellio): Modify to read secret data from file.
+			log.Infof("Found cert at %v, but is istio-generated; self-signed cert is used", fileBundle.SigningKeyFile)
+			caBundle = s.CA.GetCAKeyCertBundle().GetRootCertPem()
+			s.addStartFunc("certificate rotation", func(stop <-chan struct{}) error {
+				go func() {
+					// regenerate istiod key cert when root cert changes.
+					s.watchRootCertAndGenKeyCert(stop)
+				}()
+				return nil
```

Comment:
duplicate with above

Reply:
I didn't know if it was better to keep the conditional checks for each scenario separate to make the code more readable. This does result in duplicate code though.

Reply:
The 2 scenarios are: 1) The signing key file does not exist in the file system. 2) The signing key file exists and there is an additional istio-generated file present.

Reply:
I pushed a commit that removed the duplicate logic. The logic in the new `createSelfSignedCACertificateOptions()` doesn't need to be its own method, but I'd argue it's more readable.